sql/slave.cc

```diff
@@ -4112,6 +4112,22 @@ static int exec_relay_log_event(THD* thd, Relay_log_info* rli,
     int exec_res;
     Log_event_type typ= ev->get_type_code();
 
+    /*
+      If parallelism is enabled on the slave, we need to check if we are
+      starting after an idle state. If so, we need to update the
+      last_master_timestamp so the Seconds_Behind_Master calculation doesn't
+      compare using a timestamp from a commit long ago. This is done for
+      the scope of an entire event group, because Seconds_Behind_Master
+      should reflect the timestamp that the transaction finished.
+    */
+    if (rli->mi->using_parallel() && ev->get_type_code() == GTID_EVENT)
+    {
+      if (rli->parallel.workers_idle())
```

Comment:
This invocation deems to be too performance-wise costly.
I think we should react here on a firs event after `next_event()` wait-for-it.